packages/@sanity/portable-text-editor/src/editor/plugins/createWithPortableTextMarkModel.ts

```diff
@@ -202,14 +202,31 @@ export function createWithPortableTextMarkModel(
           editor.onChange()
         }
       }
-      // Check consistency of markDefs
+      // Check consistency of markDefs (unless we are merging two nodes)
       if (
-        isTextBlock &&
-        editor.operations.some((op) =>
-          ['split_node', 'remove_node', 'remove_text', 'merge_node'].includes(op.type),
+        editor.isTextBlock(node) &&
+        !editor.operations.some(
+          (op) => op.type === 'merge_node' && 'markDefs' in op.properties && op.path.length === 1,
```

Comment:
Because of the new normalization rules, this must be rewritten (as discovered by tests).

Now we  do the markDefs normalization here for every operation that is NOT merging two text blocks, as this will be an intermediate state, where potentially new child nodes are added later in the change (next operation). We should do this normalization in the later operations instead.